docs/resampling.ipynb

```diff
@@ -0,0 +1,380 @@
+    "\n",
+    "import scarlet\n",
+    "import scarlet.display\n",
+    "import astropy.io.fits as fits\n",
+    "from astropy.wcs import WCS\n",
+    "import sep\n",
+    "\n",
+    "%matplotlib inline\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
+    "# use a better colormap and don't interpolate the pixels\n",
+    "matplotlib.rc('image', cmap='inferno')\n",
+    "matplotlib.rc('image', interpolation='none')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load and display the sample data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load the sample data and source catalog\n",
+    "Loading the source catalog requires astropy. If you don't have astropy installed you'll have to do the source detection yourself. For example, using `sep`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "def makeCatalog(img):\n",
+    "    if np.size(img.shape) == 3:\n",
+    "        detect = img.mean(axis=0) # simple average for detection\n",
+    "    else:\n",
+    "        detect = img.byteswap().newbyteorder()\n",
+    "        \n",
+    "    bkg = sep.Background(detect)\n",
+    "    catalog = sep.extract(detect, 4, err=bkg.globalrms)\n",
+    "    if np.size(img.shape) == 3:\n",
+    "        bg_rms = np.array([sep.Background(band).globalrms for band in img])\n",
+    "    else:\n",
+    "        bg_rms =  sep.Background(detect).globalrms\n",
+    "    return catalog, bg_rms\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "obs_hdu = fits.open('../data/test_resampling/Cut_HSC.fits')\n",
+    "data_hsc = obs_hdu[0].data.byteswap().newbyteorder()\n",
+    "wcs_hsc = WCS(obs_hdu[0].header)\n",
+    "psf_hsc = fits.open('../data/test_resampling/PSF_HSC.fits')[0].data\n",
+    "\n",
+    "\n",
+    "\n",
+    "x,y = np.where(psf_hsc[0]*0==0)\n",
+    "Np1, Np2 = psf_hsc[0].shape\n",
+    "\n",
+    "hst_hdu = fits.open('../data/test_resampling/Cut_HST.fits')\n",
+    "data_hst = hst_hdu[0].data\n",
+    "wcs_hst = WCS(hst_hdu[0].header)\n",
+    "print(wcs_hst.wcs.crval)\n",
+    "wcs_hst.wcs.crval += 2.4750118475607095e-05*np.array([-np.cos(0.4136047623181346 ), -np.sin(0.4136047623181346 )])\n",
+    "print(wcs_hst.wcs.crval)\n",
+    "psf_hst = fits.open('../data/test_resampling/PSF_HST.fits')[0].data\n",
+    "np1,np2 = np.shape(psf_hst)\n",
+    "psf_hst = psf_hst.reshape(1,np1,np2)\n",
+    "\n",
+    "noise = np.concatenate((np.concatenate((data_hst[0,:], data_hst[:,0])),\n",
+    "                                            np.concatenate((data_hst[-1,:], data_hst[:,-1]))))\n",
+    "\n",
+    "n1,n2 = np.shape(data_hst)\n",
+    "data_hst = data_hst.reshape(1, n1, n2).byteswap().newbyteorder()*np.max(data_hsc)/np.max(data_hst)\n",
```

Comment:
Why are the values in `data_hst` and `data_hsc` normalized so that they have the same maximum?

Reply:
Strictly speaking, we don't need this. But let's say that one wants to make an rgb image that uses images from different datasets in the data frame, this avoids to have one band overpowering all the others.

Reply:
that's what the new `filter_weights` concept for RGB images does. In fairness, we just merged it into master yesterday ...

As a general rule, observed data should receive very little modification, and only if it's scientifically motivated